packages/plugins/htpasswd/src/htpasswd.ts

```diff
@@ -130,19 +140,19 @@ export default class HTPasswd implements IPluginAuth<HTPasswdConfig> {
    * @param {string} user
    * @param {string} password
    * @param {function} realCb
-   * @returns {function}
+   * @returns {Promise<any>}
    */
-  public adduser(user: string, password: string, realCb: Callback): any {
+  public async adduser(user: string, password: string, realCb: Callback): Promise<any> {
```

Comment:
This is a breakign change, I'd recommend use callback until there is a Promise alternative on this interface

https://github.com/verdaccio/verdaccio/blob/master/packages/core/types/index.d.ts#L578

Reply:
Well, yeah, I think still works because of the callback, but this method return a Promise and it is a bit unorhodox to return a Promise. I'd prefer have this step in the future, I've been introducing a `[method]Next` suffix to some methods to migrate progresively to aync await taste in order remove callbacks, but still fress.

![2022-03-03_21-07](https://user-images.githubusercontent.com/558752/156644572-bb2ad15e-540e-46fd-9892-07360c112d2e.png)

Reply:
I checked where this code was invoked and even though it now returns a promise, it shouldn't matter since the return value was never used. The callback that gets passed in (`realCb`) is still called as it was before. [This](https://github.com/verdaccio/verdaccio/blob/d2f9013ece8fc3b37e16e9748d03bce2acd7aba5/packages/auth/src/auth.ts#L249) is where it is called.

The TypeScript interface [`IPluginAuth`](https://github.com/verdaccio/verdaccio/blob/d2f9013ece8fc3b37e16e9748d03bce2acd7aba5/packages/core/types/index.d.ts#L576) defined for the authentication plugin also states the return type for `adduser` and other members is `void`.

Reply:
I didn't see your second comment before just posting a reply. Let me know what exactly you'd like to see change and I'll see if I can address it.

Reply:
Ah yeah, agree, return not used. Gotcha.

Reply:
It's fine, I didn't consider the `return`, we just invoke the CB, but with a comment I'm happy.